Functions.cmake

```diff
@@ -106,7 +106,10 @@ function(find_boost_libraries)
     foreach(_path ${Boost_LIBRARIES})
       if(_path MATCHES ".*boost_${_lib}\.so$")
         list(APPEND LIBS ${_path})
-      endif(_path MATCHES ".*boost_${_lib}\.so$")
+      elseif(_path MATCHES "Boost::${_lib}$")
+        # There is no path for the lib, included as a module.
+        list(APPEND LIBS ${_path})
+      endif()
```

Comment:
What is the purpose of this `find_boost_libraries()` function? I think we don't need this. We should simply use `Boost_LIBRARIES` instead of `Boost_LINK_LIBRARIES` everywhere.

Reply:
I have no idea. It was like this for some reason on the original code, I just made sure it works for newer Boost and Cmake logic too.

Reply:
I think if something has no purpose, it shouldn't be in the code. I think we can get rid of this CMake function.

Reply:
This was introduced in #260 by @mcserep, for 16.04.
I don't have a reasonable multi-boost environment to test this out nicely. If you agree this can be removed, I'll be happy to remove it... considering that linking against `Boost::foobar` most likely breaks the stuff that #260 fixed, either way...

We could introduce in the CMakeList that we need at least Boost *whichever is the stock version for 18.04 at that point* once 16.04 is unsupported? But this CMake module-target (which has nothing to do with C++20 _Modules_ ...) is only apparent with the Boost version supplied for 20.04. 18.04 still uses the "old" code.

Reply:
Another thing this function is used for is to set that only the necessary Boost libs for each end binary is linked, not *everything* from Boost that the *whole* CodeCompass needs. I.e. the `util` library does not need `boost_thread` or `boost_program_options`. If we remove this function and tell every boost-needing end binary to "link against Boost" then the project-wide `Boost COMPONENTS` list will end up being used...

Reply:
The issue was that I have multiple Boost versions installed on my development computer and had the compilation problem described in #260. I will check again, but I think I was already using Ubuntu 18.04 at that time.

Reply:
I did not manage to recreate this issue in the end (guess something changed in the last 2 years affecting the source of the problem), so we can remove this `find_boost_libraries()` function, as already proposed by #398 .